packages/client/tests/functional/relationMode/_matrix.ts

```diff
@@ -0,0 +1,64 @@
+import { defineMatrix } from '../_utils/defineMatrix'
+import { Providers } from '../_utils/providers'
+import { computeMatrix } from '../_utils/relationMode/computeMatrix'
+
+const RelationModeEnv = process.env.RELATION_MODE
+if (RelationModeEnv && RelationModeEnv !== 'prisma' && RelationModeEnv !== 'foreignKeys') {
+  throw new Error(`RELATION_MODE must be either "prisma" or "foreignKeys" but was "${RelationModeEnv}"`)
+}
+
+type RelationMode = 'prisma' | 'foreignKeys' | ''
+const relationMode: RelationMode = (RelationModeEnv as RelationMode) || ''
+
+// Note: testing 'SetDefault' requires a relation with a scalar field having the "@default" attribute.
+// If no defaults are provided for any of the scalar fields, a runtime error will be thrown.
+//
+// Note: 'Restrict' is not available when using 'sqlserver' as a provider, and it triggers a schema parsing error arising from DMMF.
+//
+// Note: 'SetNull' is only available on optional relations.
+//
+// Note: 'SetDefault' is making SQL Server crash badly
+//  const referentialActionsChoices = ['', 'Cascade', 'NoAction']
```

Comment:
?

Reply:
We do not test SetDefault at all, see https://www.notion.so/prismaio/Phase-1-Report-on-findings-f21c7bb079c5414296286973fdcd62c2#ac4d9f6a5d3842b5b6ff5b877e7e6782

Reply:
Since `SetDefault` is not valid when `relationMode="prisma"` there is no interest in doing that right now.